Add podspec.json and podfile.lock parsers

AyanSinhaMahapatra opened the pull request:
Fixes #2619 

<!--
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!
* [x] - Checked Box
* [ ] - Unchecked Box
-->

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

<!--
We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged**
if your tests fail at first!
If tests do fail, click on the red `X` to learn why by reading the logs.
Thanks!
-->

<!-- Don't forget to Signoff -->

AyanSinhaMahapatra committed:
Add podspec.json and podfile.lock parsers

Adds functions to build scancode packages from podspec.json metadata files from
cocoapods specs and also from podfile.lock files. Adds tests and testfiles.

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne commented on src/packagedcode/cocoapods.py:
```diff
@@ -170,3 +211,249 @@ def parse_person(person):
+            return self.reponame+'/issues/'
+
+    def specs_json_cdn_url(self, baseurl=default_cdn_baseurl):
+        return '{}/{}/{}/{}.podspec.json'.format(baseurl, self.name, self.version, self.name)
```
```suggestion
        return f'{baseurl}/{self.name}/{self.version}/{self.name}.podspec.json'
```

pombredanne commented on src/packagedcode/cocoapods.py:
```diff
@@ -170,3 +211,249 @@ def parse_person(person):
+    reponame = None
+    default_web_baseurl = 'https://cocoapods.org'
+    github_specs_repo_baseurl = 'https://github.com/CocoaPods/Specs/blob/master/Specs'
+    default_cdn_baseurl='https://cdn.cocoapods.org/Specs/2/2/2'
```
https://cdn.cocoapods.org/Specs/2/2/2 is likely not right and mirrors the same structure as in the repo

pombredanne commented on tests/packagedcode/data/cocoapods/podfle.lock/braintree_ios_Podfile.lock.expected.json:
```diff
@@ -0,0 +1,662 @@
+    "purl": "pkg:pods/Expecta@1.0.6",
+    "repository_homepage_url": "https://cocoapods.org/pods/Expecta",
+    "repository_download_url": null,
+    "api_data_url": "https://github.com/CocoaPods/Specs/blob/master/Specs/e/9/5/Expecta/1.0.6/Expecta.podspec.json"
```
Use the raw, direct download URL such as in https://raw.githubusercontent.com/CocoaPods/Specs/master/Specs/e/9/5/Expecta/1.0.6/Expecta.podspec.json

AyanSinhaMahapatra committed:
Add purl to csv outputs

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra committed:
Move license match functions to licensing

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

pombredanne committed:
Correct test

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne committed:
Align test results with latest code

Somehow these CSV expected results were missing a package_url and one
test needed update because of the addition of matched_rule
referenced_filenames in license scan results.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

pombredanne commented:
@AyanSinhaMahapatra FYI, I rebased and fixed the tests

pombredanne commented:
All green :green_heart:
Thanks!
Merging
